ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogWorker.java

```diff
@@ -217,8 +216,9 @@ synchronized void updateIndex(long i) {
 
     final int bufferSize = RaftServerConfigKeys.Log.writeBufferSize(properties).getSizeInt();
     this.writeBuffer = ByteBuffer.allocateDirect(bufferSize);
-    this.lastFlush = Timestamp.currentTime();
-    this.flushIntervalMin = RaftServerConfigKeys.Log.flushIntervalMin(properties);
+    this.flushExecutorWorkQueue = new ArrayBlockingQueue<>(1);
+    this.flushExecutor = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
```

Comment:
Use a 60s keepAliveTime as below -- the zero keepAliveTime probably is the reason that ExecutorService being slower than single Thread.
```
    this.flushExecutor = new ThreadPoolExecutor(1, 1, 60L, TimeUnit.SECONDS,
        this.flushExecutorWorkQueue, ConcurrentUtils.newThreadFactory(name + "-flush"));
```